nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/users/SuggestUserEntries.java

```diff
@@ -269,8 +275,20 @@ public Blob run() throws IOException {
                 user = userManager.getUserModel(userId);
                 UserAdapter userAdapter = user.getAdapter(UserAdapter.class);
                 List<String> groups = userAdapter.getGroups();
-                if (groups != null && groups.contains(groupRestriction)) {
-                    result.add(obj);
+                List<String> restrictedGroups = new ArrayList<>();
+                restrictedGroups.add(groupRestriction);
+                if (allowSubGroupsRestriction) {
+                    List<String> subGroups = userManager.getDescendantGroups(groupRestriction);
+                    for (String subGroup : subGroups) {
+                        restrictedGroups.add(subGroup);
+                    }
+                }
+
+                if (groups != null) {
+                    groups.retainAll(restrictedGroups);
+                    if (!groups.isEmpty()) {
+                        result.add(obj);
+                    }
```

Comment:
You should start by checking `groups != null` before computing the descendant groups, to avoid doing so if the user has no group.